content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,296 @@
+
+Create a `Mutation` class then initialize the mutation with the schema.
+
+```py
+class Mutation(graphene.ObjectType):
+    create_restaurant = CreateRestaurant.Field()
+```
+
+After adding the mutation and query, define the schema at the end of the `schema.py` file.
+
+```py
+schema = graphene.Schema(query=Query, mutation=Mutation)
+```
+
+Start the server and a run a mutation with the GraphQL API browser using this:
+
+```graphql
+mutation {
+    createRestaurant(name: "Kada Plaza", address: "Lekki GARDENS") {
+        ok
+        restaurant {
+            id
+            name
+            address
+        }
+    }
+}
+```
+
+The mutation returns a restaurant object with the fields that were passed in.
+
+To delete a restaurant, we would need to create a mutation. Below is the `DeleteRestaurant` mutation. Let's add it to our `schema.py` file.
+
+```python
+class DeleteRestaurant(graphene.Mutation):
+    class Arguments:
+        id = graphene.Int()
+
+    ok = graphene.Boolean()
+
+    def mutate(self, info, id):
+        restaurant = Restaurant.objects.get(id=id)
+        restaurant.delete()
+
+        return DeleteRestaurant(ok=True)
+```
+
+Next, we'll add the `DeleteRestaurant` mutation to the `Mutation` class:
+
+```py
+delete_restaurant = DeleteRestaurant.Field()
+```
+
+Run the mutation with the GraphQL API browser using this.
+
+```graphql  
+    mutation {
+        deleteRestaurant(id: 1) {
+            ok
+        }
+    }
+```
+
+Pass the restaurant id as an argument to the mutation as shown above. Output should look like this:
+
+```json
+{
+    "data": {
+        "deleteRestaurant": {
+            "ok": true
+        }
+    }
+}
+```
+
+**Note**: We should run a query to get the list of restaurants again to see the change.
+
+To update a restaurant, we would need to create a mutation. Below is the `UpdateRestaurant` mutation. We will add it to the `schema.py` file:
+```py
+class UpdateRestaurant(graphene.Mutation):
+    class Arguments:
+        id = graphene.Int()
+        name = graphene.String()
+        address = graphene.String()
+
+    ok = graphene.Boolean()
+    restaurant = graphene.Field(RestaurantType)
+
+    def mutate(self, info, id, name, address):
+        restaurant = Restaurant.objects.get(id=id)
+        restaurant.name = name
+        restaurant.address = address
+        restaurant.save()
+
+        return UpdateRestaurant(ok=True, restaurant=restaurant)
+```
+
+Let's add the `UpdateRestaurant` mutation to the `Mutation` class:
+
+```py
+update_restaurant = UpdateRestaurant.Field()
+```
+
+
+Run the mutation with the GraphQL API browser using this:
+
+``` graphql
+    mutation {
+        updateRestaurant(id: 2, name: "Kada Plaza Ltd", address: "Lekki Gardens") {
+            ok
+            restaurant {
+                id
+                name
+                address
+            }
+        }
+    }
```

Comment:
Same question here, is this `mutation` a part of a larger GraphQL structure (hence, the indentation)? If not, I suggest the following:

```suggestion
mutation {
  updateRestaurant(id: 2, name: "Kada Plaza Ltd", address: "Lekki Gardens") {
    ok
    restaurant {
      id
      name
      address
    }
  }
}
```